diff/diff.go

```diff
@@ -182,6 +182,9 @@ func diffMap(old map[string]interface{}, newAny interface{}) interface{} {
 
 // reoderKey returns the key to use for a
 func reorderKey(i interface{}) interface{} {
+	if i == nil {
```

Comment:
@jellevandenhooff per offline convo: the panic originates at `reflect.TypeOf`. If we have a typed nil (eg: `(*string)(nil)`) `reflect.TypeOf` does not panic.